controller_manager/src/controller_manager.cpp

```diff
@@ -541,34 +540,56 @@ bool ControllerManager::shutdown_controllers()
 
 void ControllerManager::init_controller_manager()
 {
-  controller_manager_activity_publisher_ =
-    create_publisher<controller_manager_msgs::msg::ControllerManagerActivity>(
-      "~/activity", rclcpp::QoS(1).reliable().transient_local());
-  rt_controllers_wrapper_.set_on_switch_callback(
-    std::bind(&ControllerManager::publish_activity, this));
-  resource_manager_->set_on_component_state_switch_callback(
-    std::bind(&ControllerManager::publish_activity, this));
+  if (!resource_manager_ && !robot_description_.empty())
+  {
+    init_resource_manager(robot_description_);
+  }
 
-  // Get parameters needed for RT "update" loop to work
-  if (is_resource_manager_initialized())
+  if (
+    is_resource_manager_initialized() && !(resource_manager_->get_joint_limiters_imported()) &&
+    params_->enforce_command_limits)
   {
-    if (params_->enforce_command_limits)
+    try
```

Comment:
why not to keep the second part of the if in the "sub-if" then you can put the message once for both for successful initialization.